### DISCUSSION TITLE
LSM6DSV driver compatibility

### DISCUSSION THREAD

**Discussion by stichkule** [zephyrproject-rtos/zephyr#72902] [Q&A] [answered]
We have a custom board based on the STM32U5A5 MCU. The board also has a LSM6DSV IMU. Looking at the zephyr driver support for this family of IMUs, there is existing support for LSM6DSV16X. Apart from the machine learning core on the 16X, are there any fundamental differences between the two variants that would prevent the use of the existing driver for LSM6DSV?

Any insights on the reusability (if even possible) of the existing driver for the LSM6DSV would be appreciated. Thanks!

**Comment by stichkule** [comment]
@avisconti Perhaps you can shed some light on this? Thanks.

**Reply by avisconti** [reply]
> @avisconti Perhaps you can shed some light on this? Thanks.

@stichkule can you possibly take alook at this answer? I marked as THE answer, but obviously you are the one expected to do it :) So, if you can spend some time to review it I'll appreciate it.

**Comment by avisconti** [comment, marked as answer]
> @avisconti Perhaps you can shed some light on this? Thanks.

The LSM6DSV and LSM6DSV16X belong to the same family of devices and are quite similar. There are few differences though. The two biggest differences are that LSM6DSV DOES NOT have MLC and QVAR, which are not currently used in zephyr. So, I would say that you may use LSM6DSv16X driver keeping in mind what I have just said. I would like to add a clear support of lsm6dsv (and lsm6dsv32x, also part of the family) in the same way we did for lps2xdf (which supports lps22df AND lps28dfw).

**Reply by stichkule** [reply]
Thanks @avisconti. I appreciate your response.

That helps clarify my question. If I understand correctly then, it is possible to use the existing LSM6DSV16X driver for the LSM6DSV, since the fundamental difference between the two variants (MLC and QVAR) are anyways not supported/used in the current zephyr driver.

**Reply by avisconti** [reply]
Yes, exactly.

**Comment by AndrewLivesChaos** [comment]
> Looking at the zephyr driver support for this family of IMUs, there is existing support for LSM6DSV16X.

Sorry, can you provide links to the documentation of LSM6DSV16X support?

**Reply by stichkule** [reply]
https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/st/lsm6dsv16x
